Allow node name reuse after delete

devernay opened the pull request:
This PR is against RB-2.4, but will be merged in RB-2.5 ultimately.

fixes #732 (this bug has been here forever)

Explanation: the nodes are not deleted while they're still in the Undo stack (which has a depth of 20 actions by default). You'll notice that if you do 20 actions (move 20 times a node in the graph), you will be able to do that rename.

I have a proposed change, which allows using the name of an existing deactivated node. To reactivate the deactivated node, the name change would be reverted by RenameNodeUndoRedoCommand::undo() anyway.

I had to add a few checks in the code:
I had to make sure that every use of Node::getScriptName_mt_safe() (except for printing messages) was also checking if the node was activated.

This seems to work.

@rodlie @YakoYakoYokuYoku Please decline the review if you think you can't review

devernay committed:
Allow node name reuse after delete

fixes #732

rodlie reviewed (approved):
Tested and works for me. I have not noticed any undo issues related to this change.

YakoYakoYokuYoku reviewed (commented):
Will test for RB-2.5 too.

devernay committed:
when creating/pasting nodes, don't add more underscores than necessary.

addresses https://github.com/NatronGitHub/Natron/issues/732#issuecomment-1000712265
